# Programming for Data Analytics - Analytics Task Two

Prepared by Samalie Piwan   
Email : spiwan@andrew.cmu.edu

In [3]:
import pandas as pd

Data Loading

In [5]:
data = pd.read_csv('penguins_size.csv')

Data Preprocessing

In [7]:
data.head(10)

species     island  culmen_length_mm  culmen_depth_mm  flipper_length_mm  \
0  Adelie  Torgersen              39.1             18.7              181.0   
1  Adelie  Torgersen              39.5             17.4              186.0   
2  Adelie  Torgersen              40.3             18.0              195.0   
3  Adelie  Torgersen               NaN              NaN                NaN   
4  Adelie  Torgersen              36.7             19.3              193.0   
5  Adelie  Torgersen              39.3             20.6              190.0   
6  Adelie  Torgersen              38.9             17.8              181.0   
7  Adelie  Torgersen              39.2             19.6              195.0   
8  Adelie  Torgersen              34.1             18.1              193.0   
9  Adelie  Torgersen              42.0             20.2              190.0   

   body_mass_g     sex  
0       3750.0    MALE  
1       3800.0  FEMALE  
2       3250.0  FEMALE  
3          NaN     NaN  
4       3450.0  FEMALE  
5       3650.0    MALE  
6       3625.0  FEMALE  
7       4675.0    MALE  
8       3475.0     NaN  
9       4250.0     NaN

**Observations**

The head() function prints out the first n rows of the dataset. Because we specified the paramter n as 10, the function above prints the first 10 rows.

This dataset contains 7 columns containing the statistics of species of penguins grouped by island. These fields are:
- species
- island
- culmen_length_mm
- culmen_depth_mm
- flipper_length_mm
- body_mass_g
- sex

We can also observe that some columns in the dataset contain missing values

In [8]:
data.tail(10)

species  island  culmen_length_mm  culmen_depth_mm  flipper_length_mm  \
334  Gentoo  Biscoe              46.2             14.1              217.0   
335  Gentoo  Biscoe              55.1             16.0              230.0   
336  Gentoo  Biscoe              44.5             15.7              217.0   
337  Gentoo  Biscoe              48.8             16.2              222.0   
338  Gentoo  Biscoe              47.2             13.7              214.0   
339  Gentoo  Biscoe               NaN              NaN                NaN   
340  Gentoo  Biscoe              46.8             14.3              215.0   
341  Gentoo  Biscoe              50.4             15.7              222.0   
342  Gentoo  Biscoe              45.2             14.8              212.0   
343  Gentoo  Biscoe              49.9             16.1              213.0   

     body_mass_g     sex  
334       4375.0  FEMALE  
335       5850.0    MALE  
336       4875.0       .  
337       6000.0    MALE  
338       4925.0  FEMALE  
339          NaN     NaN  
340       4850.0  FEMALE  
341       5750.0    MALE  
342       5200.0  FEMALE  
343       5400.0    MALE

**Observations**

The tail() function prints out the last n rows of the dataset. Because we specified the paramter n as 10, the function above prints the last 10 rows.

We can also observe that the dataset has 343 rows

In [13]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 7 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   culmen_length_mm   342 non-null    float64
 3   culmen_depth_mm    342 non-null    float64
 4   flipper_length_mm  342 non-null    float64
 5   body_mass_g        342 non-null    float64
 6   sex                334 non-null    object 
dtypes: float64(4), object(3)
memory usage: 18.9+ KB


The info() function prints out the details of the columns in  the dataset, in a range from 0 to n-1, where n is the number of columns in the dataset.

The details printed are column number, column name, length of data in the row, whether or not the column should contain null data and finally, the datatype expected in the column.

In [12]:
data.isnull().sum()

species               0
island                0
culmen_length_mm      2
culmen_depth_mm       2
flipper_length_mm     2
body_mass_g           2
sex                  10
dtype: int64

The line of code above finds the sum of the number of null values in each column. We observe that species and island have no null values, culmen_length_mm, culmen_depth_mm, flipper_length_mm and body_mass_g have 2 missing values each, and sex has 10 missing values

**b. Handle missing values:**

To handle missing data, I have opted to use imputation instead of dropping rows, to reduce information loss. Imputation is the process of replacing missing values with the mean or median value for numerical values, or the mode value for categorical values

In [23]:
# Handle missing values in culmen_length_mm, culmen_depth_mm, flipper_length_mm and body_mass_g columns using the mean value
data["culmen_length_mm"] = data["culmen_length_mm"].fillna(data["culmen_length_mm"].mean())
data["culmen_depth_mm"] = data["culmen_depth_mm"].fillna(data["culmen_depth_mm"].mean())
data["flipper_length_mm"] = data["flipper_length_mm"].fillna(data["flipper_length_mm"].mean())
data["body_mass_g"] = data["body_mass_g"].fillna(data["body_mass_g"].mean())

# Handle missing values in sex column using the mode value
data["sex"] = data["sex"].fillna(data["sex"].mode()[0])

# Check if the data still has missing values
data.isnull().sum()

species              0
island               0
culmen_length_mm     0
culmen_depth_mm      0
flipper_length_mm    0
body_mass_g          0
sex                  0
dtype: int64

**c. Check for duplicate values:**